packages/transaction-controller/src/TransactionController.ts

```diff
@@ -442,6 +452,12 @@ export class TransactionController extends BaseController<
     txParams = normalizeTxParams(txParams);
     const isEIP1559Compatible = await this.getEIP1559Compatibility();
     validateTxParams(txParams, isEIP1559Compatible);
+    await validateTransactionOrigin(
```

Comment:
We're still calling this and `getPermittedAccounts` even if `origin` isn't provided?

Shall we wrap this is a truthy check?